### ISSUE TITLE
The README example is invalid

### ISSUE THREAD

**Issue by michaelKurowski** [blainehansen/validate#1]
TS version: 5.2

The example from the README throws a TS errors:

```
@validator!!() // ⚠ Error(TS1206): Decorators are not valid here.
type Person = {
  name: string,
  age: number,
  mainLanguage: 'english' | 'spanish' | 'chinese',
  mailingAddress: null | { street: string, city: string },
  emails: { address: string, validated: boolean }[],
}


function parsePersonFile(fileContents: string) {
  const parsed: unknown = JSON.parse(fileContents)
  const personResult = Person.validator.validate(parsed)
  personResult.match({
    ok: validPerson => { console.log('yay!') },
    err: errorMessage => { console.error('boo!') }
  })
}
```

**Comment by michaelKurowski** [comment]
I've realised that it's not supposed to be a valid typescript, but a superset of it that's yet to be preprocessed by the macro library.

**Comment by blainehansen** [comment]
hahahah sorry that got you stuck though! Do you have any recommendations for documentation change to prevent that confusion from happening to anyone else?

**Comment by michaelKurowski** [comment]
Given that the ts macro library, while interesting, is not a widely used part of code processing pipeline, it would be nice to underline that the code has to be processed by the library first. When I've noticed that the macro library was mentioned, my first thought was that it's just some kind of an internal dependency. 

At first it looked like some oddly placed decorator. 

It is cool work that you're doing tho. I've found this repo because I've go interested in your projects after seeing the magmide repo. 

This repo doesn't seem to be picking much of an interest. If IDE could be somehow coerced to not see an error here, maybe it could become quite popular.
